cvmfs/magic_xattr.cc

```diff
@@ -125,21 +130,73 @@ BaseMagicXattr* MagicXattrManager::GetLocked(const std::string &name,
 void MagicXattrManager::Register(const std::string &name,
                                  BaseMagicXattr *magic_xattr)
 {
+  assert(!is_frozen_);
   if (xattr_list_.count(name) > 0) {
     PANIC(kLogSyslogErr,
           "Magic extended attribute with name %s already registered",
           name.c_str());
   }
-  magic_xattr->mount_point_ = mount_point_;
+  magic_xattr->xattr_mgr_ = this;
   xattr_list_[name] = magic_xattr;
+
+  // Mark Xattr protected so that only certain fuse_gids' can access it.
+  // If Xattr with registered "name" is part of *protected_xattrs
+  if (protected_xattrs_.count(name) > 0) {
+    magic_xattr->MarkProtected();
+  }
+}
+
+bool MagicXattrManager::IsPrivilegedGid(gid_t gid) {
+  return privileged_xattr_gids_.count(gid) > 0;
+}
+
+bool BaseMagicXattr::PrepareValueFencedProtected(gid_t gid) {
+  assert(xattr_mgr_->is_frozen());
+  if (is_protected_ && !xattr_mgr_->IsPrivilegedGid(gid)) {
+    return false;
+  }
+
+  return PrepareValueFenced();
```

Comment:
As this is not only required internally, let's make the method protected

Reply:
both `PrepareValueFencedProtected` and `PrepareValueFenced` are `public`?